### PR TITLE
[cu-206y1te, cu-26ezj25] Support time based vesting schedules in Vesting pallet

### PR DESCRIPTION
**Pull request by vimukthi-git**

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-206y1te Change vesting schedule to be able to use absolute time in addition block number](https://app.clickup.com/t/206y1te)

File: frame/composable-traits/src/vesting.rs [hunk 1]
```diff
@@ -16,56 +17,196 @@ pub trait VestedTransfer {
 		asset: Self::AssetId,
 		from: &Self::AccountId,
 		to: &Self::AccountId,
-		schedule: VestingSchedule<Self::BlockNumber, Self::Balance>,
+		schedule: VestingSchedule<Self::BlockNumber, Self::Moment, Self::Balance>,
 	) -> DispatchResult;
 }
 
+/// Vesting type for the vesting schedules.
```
Comment (vimukthi-git) [inline frame/composable-traits/src/vesting.rs:24]:
```suggestion
/// Vesting window type for the vesting schedules.
```

File: frame/composable-traits/src/vesting.rs [hunk 2]
```diff
@@ -16,56 +17,196 @@ pub trait VestedTransfer {
...
 }
 
-impl<BlockNumber: AtLeast32Bit + Copy, Balance: AtLeast32Bit + Copy>
-	VestingSchedule<BlockNumber, Balance>
+pub enum VestingWindowResult<BlockNumber, Moment> {
+	MomentResult(Moment),
+	BlockNumberResult(BlockNumber),
+}
+
+/// Impl for VestingSchedule Util with BlockNumber as the TimeMeasure
```
Comment (vimukthi-git) [inline frame/composable-traits/src/vesting.rs:61]:
```suggestion
```

File: frame/composable-traits/src/vesting.rs [hunk 3]
```diff
@@ -16,56 +17,196 @@ pub trait VestedTransfer {
...
 		self.per_period
 			.checked_mul(&unrealized.into())
 			.expect("ensured non-overflow total amount; qed")
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use crate::vesting::VestingWindow::{BlockNumberBased, MomentBased};
+	// Note this useful idiom: importing names from outer (for mod tests) scope.
```
Comment (vimukthi-git) [inline frame/composable-traits/src/vesting.rs:129]:
```suggestion
```

**Review by hussein-aitlahcen — APPROVED** [review]
Well done :)

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by vivekvpandya — APPROVED** [review]

**Pull request merged**